File: lab/ki305/kohut/lab3/Frigate.java
package ki305.kohut.lab3;

/**
 * Клас Frigate представляє фрегат з озброєнням, наслідує базовий клас Ship.
 */
public class Frigate extends Ship implements Armed {
    private int weaponCount; // Кількість зброї на борту

    public Frigate(String name, int displacement, int speed, int weaponCount) {
        super(name, displacement, speed);
        this.weaponCount = weaponCount;
    }

    @Override
    public void fireWeapon() {
        if (weaponCount > 0) {
            System.out.println("Фрегат " + name + " здійснив постріл!");
            weaponCount--;
        } else {
            System.out.println("У фрегата " + name + " немає боєприпасів!");
        }
    }

    @Override
    public void reloadWeapons(int ammo) {
        weaponCount += ammo;
        System.out.println("Фрегат " + name + " поповнив боєприпаси. Кількість зброї: " + weaponCount);
    }
}

File: lab/ki305/kohut/lab9/Frigate.py
# Frigate.py
from Ship import Ship

class Frigate(Ship):
    def __init__(self, name, length, speed, armament):
        """
        Ініціалізує фрегат із заданими характеристиками.

        :param name: Назва корабля
        :param length: Довжина корабля (м)
        :param speed: Швидкість корабля (вузли)
        :param armament: Список озброєння
        """
        super().__init__(name, length, speed)  # Виклик конструктора базового класу
        self.armament = armament  # Озброєння фрегата

    def add_weapon(self, weapon):
        """
        Додає зброю до списку озброєння.

        :param weapon: Назва зброї
        """
        self.armament.append(weapon)

    def get_armament(self):
        """
        Повертає список озброєння.
        """
        return f"Озброєння {self.name}: {', '.join(self.armament)}"
